### PR TITLE
Remove unneeded extra chars to reduce search-index size

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Before:

```
2013782 Dec 11 10:16 build/x86_64-unknown-linux-gnu/doc/search-index.js
```

After:

```
1736597 Dec 11 10:50 build/x86_64-unknown-linux-gnu/doc/search-index.js
```

No changes in the output of the search.

r? @QuietMisdreavus

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
The `plain_summary_line` function is not just used for the search index. It's also used for item listings on module pages, so you're effectively truncating those listings as well.

You're also not accounting for combined characters, so this will cut between a letter and a combined accent if it happens to land on that boundary.

Since this is going to be used for user-displayed text, i'd prefer it if this used a smarter means of collecting characters (maybe use `split_whitespace` and count codepoints as you add words to the final string?) and ended with a "..." if we're cutting it off.

**Comment by GuillaumeGomez** [comment]
Arf, I thought combined accent was accounted for one char in Rust (which is why I did it this way...). I'll add a parameter to only do it when required.

**Comment by QuietMisdreavus** [comment]
Nah, `chars()` only splits at codepoints, it doesn't account for what a printable "character" is. So something with combining characters, where multiple codepoints converge on the same screen location, will still have multiple items in that iterator. (If you want to step up to printable "graphemes", that's a whole other crate, `unicode-segmentation` - but since this is mainly a file-size optimization and not a screen-size one, i'm not sure it will be worth bringing in the new crate. It is worth considering that this is a sort of penalty for languages with lots combining characters, though.)

**Comment by GuillaumeGomez** [comment]
Then I can just split on whitespaces and as long as we're under the limit, we'll add the word.

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/librustdoc/html/render.rs [hunk 1]
```diff
@@ -2396,7 +2396,13 @@ fn shorter<'a>(s: Option<&'a str>) -> String {
 #[inline]
 fn plain_summary_line(s: Option<&str>) -> String {
     let line = shorter(s).replace("\n", " ");
-    markdown::plain_summary_line(&line[..])
+    markdown::plain_summary_line_full(&line[..], false)
+}
+
+#[inline]
+fn plain_summary_line_full(s: Option<&str>) -> String {
```
Comment (QuietMisdreavus) [inline src/librustdoc/html/render.rs]: I feel like the naming of this is off - this is the version that *truncates* the summary line, not the one that gives the full version. This should be named something like `plain_summary_line_short`.

Also, if you want to make the aliases file also use this version, you can change the call on line 1682 to use this.

**Comment by GuillaumeGomez** [comment]
Updated.

**Review by QuietMisdreavus — APPROVED** [review]
All looks good now, thanks so much!

**Comment by QuietMisdreavus** [comment]
r=me when travis is green.

**Pull request merged**